src/OrbitGl/BatcherTest.cpp

```diff
@@ -237,4 +241,43 @@ TEST(Batcher, MultipleDrawCalls) {
   EXPECT_DEATH((void)batcher.GetUserData(id), "size");
 }
 
+bool LineEq(const Line& lhs, const Line& rhs) {
+  return lhs.start_point == rhs.start_point && lhs.end_point == rhs.end_point;
+}
+
+TEST(Batcher, TranslationsAreAutomaticallyAdded) {
+  MockBatcher batcher(BatcherId::kUi);
+  batcher.AddLine(Vec2(0.f, 0.f), Vec2(1.f, 1.f), 0.f, Color());
+
+  const PrimitiveBuffers& buffers = batcher.GetInternalBuffers(0.f);
+  Line original_expectation{Vec3(0.f, 0.f, 0.f), Vec3(1.f, 1.f, 0.f)};
+  Vec3 transform(10.f, 100.f, 0.1f);
+  Line transformed_expectation{original_expectation.start_point + transform,
+                               original_expectation.end_point + transform};
+  auto it = buffers.line_buffer.lines_.begin();
+
+  ASSERT_TRUE(LineEq(original_expectation, *it));
+
+  batcher.PushTranslation(10, 100, 0.1f);
+  // Should not affect previously added lines
+  ASSERT_TRUE(LineEq(original_expectation, *it));
+
+  batcher.AddLine(Vec2(0.f, 0.f), Vec2(1.f, 1.f), 0.f, Color());
+  ++it;
+  ASSERT_TRUE(LineEq(transformed_expectation, *it));
+  batcher.PushTranslation(0, 0, 0.f);
+
+  batcher.AddLine(Vec2(0.f, 0.f), Vec2(1.f, 1.f), 0.f, Color());
+  ++it;
+  ASSERT_TRUE(LineEq(transformed_expectation, *it));
+
+  batcher.PopTranslation();
+  batcher.PopTranslation();
```

Comment:
Since you are doing a nice and complete test here, why not adding an additional check in between PopTranslations to check they are being erased from the newest to the oldest?

Reply:
Since I'm pushing a (0, 0) transform, there shouldn't be a difference before and after the first pop. But I've added an explicit check for this, that seems useful.